src/utils/error_utils.go

```diff
@@ -0,0 +1,45 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+)
+
+type AssistedServiceError interface {
+	Error() string
+	GetPayload() *models.Error
+}
+
+type AssistedServiceInfraError interface {
+	Error() string
+	GetPayload() *models.InfraError
+}
+
+func BuildMessageFromAssistedServiceError(err error) string {
```

Comment:
I think that this function should return an error then you will not need the next function `WrappedAssistedServiceErrorMessage` because you will be able to use the errors package to wrap an error.  
Then the name should probably change as well, `GetAssistedError` or something like this.